blame.go

```diff
@@ -0,0 +1,128 @@
+package git
+
+import (
+	"bufio"
+	"bytes"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// BlameFile return map of line number and Commit that change that line
+func (r *Repository) BlameFile(file string) (map[int]*Commit, error) {
```

Comment:
The design of this API should look like:

https://github.com/gogs/git-module/blob/0bb10af0874d5c74f0967bb4b2d0102d11341404/repo_diff.go#L26


For example:

```go
func (r *Repository) Blame(rev, file string, opts ...BlameOptions) (*Blame, error) {
```